### ISSUE TITLE
String Representation of an Enum

### ISSUE THREAD

**Issue by jaredbroad** [pythonnet/pythonnet#575]
### Environment

-   Pythonnet version: All
-   Python version: 2.7
-   Operating System: Mono

### Details
-   Print string name of an enum from python.
```python   
   // in C#
   enum NamePrinting {
       Test,
       Name
   }

   // in python:
   self.A = "A" + str(NamePrinting.Test)
   print A
   // A1 not "ATest" like in C#.
```

Is it possible to get the text string name instead of the integer?

**Comment by denfromufa** [comment]
@jaredbroad the problem here is that C# enum is converted to its value (int) before you can do anything with this enum in Python. Here are possible fixes for similar issue, that I suggested before:

https://github.com/pythonnet/pythonnet/issues/174#issuecomment-192835858

**Comment by aparamon** [comment]
Current `str()` representation of .NET enums is sub-optimal per se (strong .NET typing information is lost) and also is an incompatibility with IronPython.

@denfromufa suggested several approaches to support better enum `str()` representation, including
* special case enums in overloading method selection to allow inter-operability with integer types.

What about using [IntEnum](https://docs.python.org/3/library/enum.html#intenum)? According to the docs, special casing enums may not be at all required as `IntEnum`s are already `int`s.

**Comment by denfromufa** [comment]
@aparamon this would work only in Python 3 which is ok. This contribution is welcome!

**Comment by scenapp** [comment]
Enums are enums in .NET - this is a very weird decision to replace with integers. Are we back to the 'error code days' ?

**Comment by filmor** [comment]
Contributions are welcome. Python.NET is a very old library, dating back to Python 2.2 (I think), `IntEnum` etc. didn't exist back then.

**Comment by MarioQuillas** [comment]
I tried to replicate this issue because I'd like to contribute but this seems to work for .net versions greater than 4.6

**Comment by filmor** [comment]
This was indeed implemented by @lostmsu in #1392 (so it works with builds from the current `master`, independent of the .NET version).
